services/app-api/src/resolvers/user/updateCMSUser.ts

```diff
@@ -39,56 +43,81 @@ export function updateCMSUserResolver(
+                    extensions: {
+                        code: 'BAD_USER_INPUT',
+                        cause: 'INVALID_DIVISION_ASSIGNMENT',
+                    },
                 })
             }
         }
 
         const stateAssignmentCodes: StateCodeType[] = []
         const invalidCodes = []
+        let invalidStateCodes
 
-        for (const assignment of stateAssignments) {
-            if (isValidStateCode(assignment)) {
-                stateAssignmentCodes.push(assignment)
-            } else {
-                invalidCodes.push(assignment)
+        if (stateAssignments) {
+            for (const assignment of stateAssignments) {
+                if (isValidStateCode(assignment)) {
+                    stateAssignmentCodes.push(assignment)
+                } else {
+                    invalidCodes.push(assignment)
+                }
             }
-        }
 
-        // check that the state codes are valid
-        if (invalidCodes.length > 0) {
-            const invalidStateCodes = stateAssignments.filter(
-                (assignment) => !isValidStateCode(assignment)
-            )
+            // check that the state codes are valid
+            if (invalidCodes.length > 0) {
+                invalidStateCodes = stateAssignments.filter(
+                    (assignment) => !isValidStateCode(assignment)
+                )
 
-            const errMsg = 'Invalid state codes'
-            logError('updateCmsUser', errMsg)
-            setErrorAttributesOnActiveSpan(errMsg, span)
-            throw new UserInputError(errMsg, {
-                argumentName: 'stateAssignments',
-                argumentValues: invalidStateCodes,
-            })
+                const errMsg = 'Invalid state codes'
+                logError('updateCmsUser', errMsg)
+                setErrorAttributesOnActiveSpan(errMsg, span)
+                throw new UserInputError(errMsg, {
+                    argumentName: 'stateAssignments',
+                    argumentValues: invalidStateCodes,
+                    code: 'BAD_USER_INPUT',
+                    cause: 'INVALID_STATE_CODES',
+                })
+            }
         }
-
         const result = await store.updateCmsUserProperties(
             cmsUserID,
             stateAssignmentCodes,
-            divisionAssignment
+            currentUser.id,
+            divisionAssignment,
+            'Updated user assignments' // someday might have a note field and make this a param
         )
         if (isStoreError(result)) {
             if (result.code === 'INSERT_ERROR') {
                 const errMsg = 'cmsUserID does not exist'
                 logError('updateCmsUser', errMsg)
                 setErrorAttributesOnActiveSpan(errMsg, span)
-                throw new UserInputError(errMsg, { argumentName: 'cmsUserID' })
+                throw new UserInputError(errMsg, {
+                    argumentName: 'cmsUserID',
+                    code: 'BAD_USER_INPUT',
```

Comment:
I think this one needs to be in `extensions: {` still

Reply:
Looking at the docs, it seems that `extensions: {` is an error _output_ convention, and not used while throwing.  I think  we can add whatever fields we want to the object passed in as the second parameter.

https://www.apollographql.com/docs/apollo-server/v3/data/errors/#throwing-errors

I'll hold off merging in case @JasonLin0991 knows something I don't know, otherwise I'll update the error handling doc as part of this PR, to make this clear.

Reply:
I believe when using error classes from `apollo-server-lambda`, the second parameter is the `extensions` object. These error classes also already set the error codes so in this case `UserInputError` is already set to `BAD_USER_INPUT` and the extra `code:  BAD_USER_INPUT` is not needed.

From `apollo-server-lambda`
```typescript
export class UserInputError extends ApolloError {
  constructor(message: string, extensions?: Record<string, any>) {
    super(message, 'BAD_USER_INPUT', extensions);

    Object.defineProperty(this, 'name', { value: 'UserInputError' });
  }
}
```

Reply:
Yeah, we agree.  I think this, in the error handling doc, is not right though.  What do you think @JasonLin0991?
```typescript
throw new GraphQLError('Email failed.', {
    extensions: {
        code: 'INTERNAL_SERVER_ERROR',
        cause: 'DB_ERROR',
    },
})
```